### ISSUE TITLE
ExecuteUpdate for Dynamic Entity

### ISSUE THREAD

**Issue by aslooni** [dotnet/efcore#33626]
I am trying to use the **ExecuteUpdate** method with a dynamic entity. When using the following code snippet where my entity is dynamic, I encounter the following error:

```
var parameterExpression = Expression.Parameter(typeof(T), "x"); 
var memberExpression = Expression.Property(parameterExpression, "MyProp"); 
var conversion = Expression.Convert(memberExpression, typeof(object)); 
var lambda = Expression.Lambda<Func<T, object>>(conversion, parameterExpression).Compile();
await query.ExecuteUpdateAsync(c => c.SetProperty(lambda, newValue));
```

Error message: 
`Unable to cast object of type 'System.Linq.Expressions.TypedParameterExpression' to type 'System.Linq.Expressions.LambdaExpression'.`

How can I resolve this issue?

**Comment by vigouredelaruse** [comment]
because all those lambda expressions end up looking like greek to me, i wrote an extension method based on this
https://stackoverflow.com/questions/47673524/ef-core-soft-delete-with-shadow-properties-and-query-filters/48744644#48744644
```
        public static LambdaExpression ConvertFilterExpression<TInterface>(
               this Expression<Func<TInterface, bool>> filterExpression,
               Type entityType)
        {
            var newParam = Expression.Parameter(entityType);
            var newBody = ReplacingExpressionVisitor.Replace(filterExpression.Parameters.Single(), newParam, filterExpression.Body);

            return Expression.Lambda(newBody, newParam);
        }
```
then i write the prediates in plain english like this
```
Expression<Func<IContentRowLevelSecured, bool>> newFilter = f =>
// filter everyone deny
adminPredicate(this.claimsPrincipal) || !f.AccessControlEntries
    .Where(w => w.Scope == ACEPermissionScope.EVERYONE
    && w.Permission == ACEPermission.READ
    && w.PermissionType == ACEPermissionType.DENY
    && w.IsActive == true).Any()
// filter anonymous deny for logged in scnario
|| (adminPredicate(this.claimsPrincipal) || !f.AccessControlEntries
    .Where(w => w.Scope == ACEPermissionScope.ANONYMOUS
    && w.Permission == ACEPermission.READ
    && w.PermissionType == ACEPermissionType.DENY
    && w.IsActive == true).Any())
// filter explict deny to current user
&& !f.AccessControlEntries 
// etc
```

then 
```
                                    var newFilterLambda = newFilter.ConvertFilterExpression<IContentRowLevelSecured>(entity.ClrType);
                                    newFilterLambda.Compile();

// now ef core will let you use the lambda in a global query filter for example
  builder.Entity(entity.ClrType).AddQueryFilter(newFilter);
```

καλή τύχη

**Comment by roji** [comment]
@aslooni am going to close this as a usage issue, and the above should get you what you need (though it's certainly more complex than it should be).
